File: resolucoes/kotlin/lista01/src/main/kotlin/ex008.kt
import java.util.Scanner

/**
 * Desenvolva um programa que leia uma distância em metros e mostre os valores
 * relativos em outras medidas.
 * Ex:
 * Digite uma distância em metros: 185,72
 * A distância de 85,7 m corresponde a:
 * 0,18572 km 1,8572 hm 18.572 dam
 * 1857,2 dm 18572,0 cm 185720,0 mm
 * */
fun ex008(){
    val scanner = Scanner(System.`in`)
    print("Digite uma distância em metros: ")
    val metros = scanner.nextDouble()
    print("""
    A distância de ${metros}m corresponde a:
    ${metros / 1000}km   ${metros / 100}hm     ${metros / 10}dam
    ${metros * 10}dm   ${metros * 100}cm   ${metros * 1000}mm
    """.trimIndent())
}

internal fun main(){
    ex008()
}

File: resolucoes/php/lista01/ex008.php
<?php
/*
 * Desenvolva um programa que leia uma distância em metros e mostre os valores relativos em outras medidas. ex:
 *
 * Digite uma distância em metros: 185.72
  A distância de 85.7m corresponde a:
  0.18572Km           1857.2dm
  1.8572Hm           18572.0cm
  18.572Dam           185720.0mm
 */

$inputValue = readline("Digite uma distância em metros: ");

if (preg_match('/^[+-]?\d*\.?\d+$/', $inputValue)) {
  $number = floatval($inputValue);
  $km =  $number / 1000;
  $hm = $number  /100;
  $dam = $number / 10;
  $dm = $number * 10;
  $cm = $number * 100;
  $mm = $number *1000;

  echo "A distãncia de " . number_format($number,1,".",","). "m corresponde a: ";

  print "\nKilometer    : " . $km;
  print "\nDecametro    : " . $dam;
  print "\nCentimetro   : " . $cm;
  print "\nHectometro   : " . $hm;
  print "\nDecimeter    : " . $dm;
  print "\nMilimetro    : " . $mm;
}


